Umbrella: strengthen unit test coverage across AgentCube components

## Summary
- Expand unit test coverage across AgentCube control plane, data plane, and tooling.
- Establish baseline coverage targets (>=70% for Go packages, smoke coverage for Python CLI/SDK) with CI reporting.

## Motivation
- Protect the rapidly evolving proposal codebase from regressions as we bring components online.
- Build confidence for future contributors and reviewers by providing fast feedback loops.

## Goals
- Capture current coverage numbers and document the biggest gaps.
- Add focused unit tests for critical behaviors across the components listed below.
- Wire coverage reporting into CI (Go and Python) so we can track improvements over time.

## Workstreams
### Go control plane & services
- [ ] `pkg/agentd`: cover runtime registration, lifecycle orchestration, and error paths.
- [ ] `pkg/picod`: exercise execution pipeline, auth helpers, and filesystem utilities.
- [ ] `pkg/router`: validate session manager, HTTP/WebSocket handlers, and config parsing.
- [ ] `pkg/workloadmanager`: focus on sandbox/code interpreter controllers, auth/JWT helpers, and workload builder.
- [ ] `pkg/store`: ensure interface contract tests plus Redis/Valkey adapters (consider fake clients or testcontainers).
- [ ] `pkg/common/types` & shared utilities: add serialization/validation tests where missing.

### Binaries & entrypoints
- [ ] `cmd/agentd`, `cmd/router`, `cmd/workload-manager`, `cmd/picod`: ensure flag parsing and wiring logic have basic coverage (e.g., using in-memory config).

### CLI & Python SDK
- [ ] `cli/agentcube`: add pytest coverage for command behaviors and error handling.
- [ ] `sdk-python`: cover agentcube clients/utilities with mocked HTTP interactions.

### Tooling & CI
- [ ] Update `Makefile`/CI to run `go test ./... -coverprofile` and surface coverage summary.
- [ ] Add Python test job (pytest) with coverage report for CLI/SDK.
- [ ] Document how to run the test suites locally (README or docs/devguide) and set initial coverage targets.

## Exit Criteria
- Checklist above is tackled via focused follow-up issues/PRs linked back here.
- Coverage targets and CI jobs are in place, with documentation updated accordingly.

Please link follow-up issues/PRs to this umbrella to track progress.

running go test ./... -cover 

```
        github.com/volcano-sh/agentcube/cmd/agentd              coverage: 0.0% of statements
        github.com/volcano-sh/agentcube/cmd/picod               coverage: 0.0% of statements
        github.com/volcano-sh/agentcube/cmd/router              coverage: 0.0% of statements
        github.com/volcano-sh/agentcube/cmd/workload-manager            coverage: 0.0% of statements
--- FAIL: TestReconciler_Reconcile_WithLastActivity (0.20s)
    --- FAIL: TestReconciler_Reconcile_WithLastActivity/Non-running_sandbox_should_not_be_processed (0.00s)
        agentd_test.go:142: 
                Error Trace:    /root/go/src/agent-box/pkg/agentd/agentd_test.go:142
                Error:          Received unexpected error:
                                sandboxs.agents.x-k8s.io "test-sandbox" not found
                Test:           TestReconciler_Reconcile_WithLastActivity/Non-running_sandbox_should_not_be_processed
FAIL
coverage: 65.0% of statements
FAIL    github.com/volcano-sh/agentcube/pkg/agentd      0.237s
        github.com/volcano-sh/agentcube/pkg/apis/runtime/v1alpha1               coverage: 0.0% of statements
        github.com/volcano-sh/agentcube/pkg/common/types                coverage: 0.0% of statements
ok      github.com/volcano-sh/agentcube/pkg/picod       1.485s  coverage: 65.2% of statements
ok      github.com/volcano-sh/agentcube/pkg/router      17.003s coverage: 88.8% of statements
ok      github.com/volcano-sh/agentcube/pkg/store       0.041s  coverage: 70.7% of statements
ok      github.com/volcano-sh/agentcube/pkg/workloadmanager     2.164s  coverage: 7.5% of statements
```

> running go test ./... -cover运行 go test ./... -cover
> 
> ```
>         github.com/volcano-sh/agentcube/cmd/agentd              coverage: 0.0% of statements
>         github.com/volcano-sh/agentcube/cmd/picod               coverage: 0.0% of statements
>         github.com/volcano-sh/agentcube/cmd/router              coverage: 0.0% of statements
>         github.com/volcano-sh/agentcube/cmd/workload-manager            coverage: 0.0% of statements
> --- FAIL: TestReconciler_Reconcile_WithLastActivity (0.20s)
>     --- FAIL: TestReconciler_Reconcile_WithLastActivity/Non-running_sandbox_should_not_be_processed (0.00s)
>         agentd_test.go:142: 
>                 Error Trace:    /root/go/src/agent-box/pkg/agentd/agentd_test.go:142
>                 Error:          Received unexpected error:
>                                 sandboxs.agents.x-k8s.io "test-sandbox" not found
>                 Test:           TestReconciler_Reconcile_WithLastActivity/Non-running_sandbox_should_not_be_processed
> FAIL
> coverage: 65.0% of statements
> FAIL    github.com/volcano-sh/agentcube/pkg/agentd      0.237s
>         github.com/volcano-sh/agentcube/pkg/apis/runtime/v1alpha1               coverage: 0.0% of statements
>         github.com/volcano-sh/agentcube/pkg/common/types                coverage: 0.0% of statements
> ok      github.com/volcano-sh/agentcube/pkg/picod       1.485s  coverage: 65.2% of statements
> ok      github.com/volcano-sh/agentcube/pkg/router      17.003s coverage: 88.8% of statements
> ok      github.com/volcano-sh/agentcube/pkg/store       0.041s  coverage: 70.7% of statements
> ok      github.com/volcano-sh/agentcube/pkg/workloadmanager     2.164s  coverage: 7.5% of statements
> ```

TestReconciler_Reconcile_WithLastActivity/Non-running_sandbox_should_not_be_processed verifies that non-running sandboxes (with ConditionReady == False) are not processed or deleted, even if their last activity is expired.
The current Reconcile method processes all sandboxes regardless of status. The test expects only running sandboxes to be processed.
Should we modify agentd.go to add a running status check before processing, so only sandboxes with ConditionReady == True are handled, or should we modify the tests?

--- a/pkg/api/errors_test.go
+++ b/pkg/api/errors_test.go
@@ -46,38 +46,6 @@ func TestNewSessionNotFoundError(t *testing.T) {
 	assert.Equal(t, sessionID, status.Details.Name)
 }
 
-func TestNewSessionNotFoundError_DifferentSessionIDs(t *testing.T) {
-	tests := []struct {
-		name     string
-		sessionID string
-	}{
-		{
-			name:      "simple session ID",
-			sessionID: "session-1",
-		},
-		{
-			name:      "UUID session ID",
-			sessionID: "550e8400-e29b-41d4-a716-446655440000",
-		},
-		{
-			name:      "empty session ID",
-			sessionID: "",
-		},
-		{
-			name:      "long session ID",
-			sessionID: "very-long-session-id-with-many-characters-that-should-still-work",
-		},
-	}
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			err := NewSessionNotFoundError(tt.sessionID)
-			assert.Error(t, err)
-			assert.True(t, apierrors.IsNotFound(err))
-		})
-	}
-}
-
 func TestWorkloadResource(t *testing.T) {
 	tests := []struct {
 		name     string
@@ -167,19 +135,6 @@ func TestNewSandboxTemplateNotFoundError(t *testing.T) {
 	}
 }
 
-func TestNewSandboxTemplateNotFoundError_ErrorFormat(t *testing.T) {
-	namespace := "test-namespace"
-	name := "test-resource"
-	kind := types.AgentRuntimeKind
-
-	err := NewSandboxTemplateNotFoundError(namespace, name, kind)
-
-	// Verify error message contains expected information
-	errMsg := err.Error()
-	assert.Contains(t, errMsg, namespace)
-	assert.Contains(t, errMsg, name)
-}
-
 func TestNewUpstreamUnavailableError(t *testing.T) {
 	tests := []struct {
 		name    string
@@ -208,7 +163,7 @@ func TestNewUpstreamUnavailableError(t *testing.T) {
 			if tt.wantErr {
 				// nil error will cause panic when calling err.Error()
 				assert.Panics(t, func() {
-					NewUpstreamUnavailableError(tt.wrapped)
+					_ = NewUpstreamUnavailableError(tt.wrapped)
 				})
 				return
 			}
@@ -257,7 +212,7 @@ func TestNewInternalError(t *testing.T) {
 			if tt.wantErr {
 				// nil error will cause panic when calling err.Error()
 				assert.Panics(t, func() {
-					NewInternalError(tt.wrapped)
+					_ = NewInternalError(tt.wrapped)
 				})
 				return
 			}
@@ -293,20 +248,6 @@ func TestNewInternalError_ErrorWrapping(t *testing.T) {
 	// Instead, the error is accessible through the StatusError interface
 	// Check that the error message contains the original error
 	assert.Contains(t, wrappedErr.Error(), originalErr.Error())
-}
-
-func TestErrorConstants(t *testing.T) {
-	// Verify error constants are defined
-	assert.NotNil(t, ErrAgentRuntimeNotFound)
-	assert.NotNil(t, ErrCodeInterpreterNotFound)
-	assert.NotNil(t, ErrTemplateMissing)
-	assert.NotNil(t, ErrPublicKeyMissing)
-
-	// Verify error messages
-	assert.Contains(t, ErrAgentRuntimeNotFound.Error(), "agent runtime")
-	assert.Contains(t, ErrCodeInterpreterNotFound.Error(), "code interpreter")
-	assert.Contains(t, ErrTemplateMissing.Error(), "template")
-	assert.Contains(t, ErrPublicKeyMissing.Error(), "public key")
 }
 
 func TestNewSandboxTemplateNotFoundError_ResourceGroup(t *testing.T) {

